pkgs/shells/zsh/zimfw/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zimfw";
+  version = "v1.11.0";
+  src = fetchFromGitHub {
+    owner = "zimfw";
+    repo = pname;
+    rev = version;
+    ## zim only needs this one file to be installed.
+    sparseCheckout = [ "zimfw.zsh" ];
+    sha256 = "sha256-BmzYAgP5Z77VqcpAB49cQLNuvQX1qcKmAh9BuXsy2pA=";
+  };
+  strictDeps = true;
+  dontConfigure = true;
+  dontBuild = true;
+  dontPatch = true;
```

Comment:
```suggestion
```

The other two are fine, but is there any compelling reason to set `dontPatch = true`? (especially if, e.g., someone overrides to add a patch file)

Reply:
tbh I based everything on `../zplug/default.nix`. I'm happy to apply all of your suggestions.

Reply:
I think the `repo = pname` one isn't that bad if you really want to leave it, but all of the other changes probably should be applied (and zplug was added a few years ago so some best practices have been changed or more strictly enforced)